trackersite/tracker/views.py

```diff
@@ -556,19 +666,629 @@ def cluster_detail(request, pk):
+                            real = user.paid_expeditures()
+                            if real >= larger:
+                                tmp.append(user)
+                        users = tmp
+                        del(tmp)
+                    elif larger == '' and smaller != '':
+                        smaller = int(smaller)
+                        tmp = []
+                        for user in users:
+                            real = user.paid_expeditures()
+                            if real <= smaller:
+                                tmp.append(user)
+                            users = tmp
+                            del(tmp)
+
+                    if 'user-permision' in request.POST:
+                        priv = request.POST['user-permision']
+                        tmp = []
+                        if priv == 'normal':
+                            for user in users:
+                                if not user.user.is_staff and not user.user.is_superuser:
+                                    tmp.append(user)
+                        elif priv == 'staff':
+                            for user in users:
+                                if user.user.is_staff:
+                                    tmp.append(user)
+                        elif priv == 'superuser':
+                            for user in users:
+                                if user.user.is_superuser:
+                                    tmp.append(user)
+                        else:
+                            return HttpResponseBadRequest('You must fill the form validly')
+                        users = tmp
+                        del(tmp)
+
+                    response = HttpResponseCsv(['id', 'username', 'first_name', 'last_name', 'email', 'is_active', 'is_staff', 'is_superuser', 'last_login', 'date_joined', 'created_tickets', 'accepted_expeditures', 'paid_expeditures', 'bank_account', 'other_contact', 'other_identification'])
+                    response['Content-Disposition'] = 'attachment; filename="exported-users.csv"'
+                    for user in users:
+                        response.writerow([user.user.id, user.user.username, user.user.first_name, user.user.last_name, user.user.email, user.user.is_active, user.user.is_staff, user.user.is_superuser, user.user.last_login, user.user.date_joined, user.count_ticket_created(), user.accepted_expeditures(), user.paid_expeditures(), user.bank_account, user.other_contact, user.bank_account])
+                    return response
+            return HttpResponseForbidden(_('You must be staffer in order to export users'))
+
+        return HttpResponseBadRequest(_('You must fill the form validly'))
+    else:
+        return render(
+            request,
+            'tracker/export.html',
+            {
+                'topics': Topic.objects.all(),
+                'users': User.objects.all(),
+                'tickets': Ticket.objects.all(),
+            })
```

Comment:
Tohle určitě jde napsat líp než jako funkci o pěti stech řádcích.

Reply:
Pokusím se vymyslet, jak.